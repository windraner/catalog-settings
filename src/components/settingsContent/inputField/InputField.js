import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import './inputField.css';

export default class InputField extends Component {
  state={
    isValidated: false,
  }

  changeHandler = (e) => {
    const { name } = this.props;
    this.props.onChange(name, e.target.value);
    if(e.target.value.trim()) {
      this.setState({isValidated: false});
    }
  }

  blurHandler = (e) => {
    const { name } = this.props;
    this.props.onChange(name, e.target.value.trim());
    if(!e.target.value.trim()) {
      this.setState({isValidated: true});
    }
  }

  renderError = () => {
    const { required } = this.props;
    const { isValidated } = this.state;
    if(required && isValidated) {
      return (<div className="catalog-settings-content__error-message">Can not be empty</div>);
    }
    return null;
  }

  render() {
    const { value, required } = this.props;
    const { isValidated } = this.state;
    const className = (required && isValidated ?
      'catalog-settings-content__input-field catalog-settings-content__input-field_error'
      :
      'catalog-settings-content__input-field');

    return (
      <Fragment>
        <input
          value={value ? value : ''}
          onChange={this.changeHandler}
          onBlur={this.blurHandler}
          type="text"
          className={className}
        />
        {this.renderError()}
      </Fragment>
    );
  }
}

InputField.propTypes = {
  value: PropTypes.string,
  required: PropTypes.bool
};
