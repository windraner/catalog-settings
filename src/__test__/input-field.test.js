import React from 'react';
import InputField from '../components/settings-content/input-field/input-field';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<InputField />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<InputField value="test" />);
    expect(wrapper.find('.catalog-settings-content__input-field').length).toBe(1);
    expect(wrapper.find('.catalog-settings-content__input-field').props().value).toBe('test');
  });
  it('Check validation', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper.find('.catalog-settings-content__error-message').length).toBe(0);
    wrapper.setState({ isValidated: true });
    expect(wrapper.find('.catalog-settings-content__error-message').length).toBe(0);
    wrapper.setProps({ required: true });
    expect(wrapper.find('.catalog-settings-content__error-message').text()).toBe('Can not be empty');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<InputField />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
