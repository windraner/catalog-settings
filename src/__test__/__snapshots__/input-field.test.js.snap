// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> matches the snapshot 1`] = `
<Fragment>
  <input
    className="catalog-settings-content__input-field d-block w-100 mt-1"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
    value=""
  />
</Fragment>
`;
